### PR TITLE
Added Remove label functionality and updated documentation and bug fixes

### PR DESCRIPTION
**Pull request by mahemaku**

### CODE REVIEW

File: data_labeling_examples/bulk_labeling_java/README.md [hunk 1]
```diff
@@ -125,32 +158,32 @@ java -version
...
 ```
 6. Run the below command to bulk label by "FIRST_LETTER_MATCH" labeling algorithm.
 
 ```
-java -DCONFIG_FILE_PATH='~/.oci/config' -DCONFIG_PROFILE=DEFAULT -DDLS_DP_URL=https://dlsprod-dp.us-ashburn-1.oci.oraclecloud.com -DTHREAD_COUNT=20 -DDATASET_ID=ocid1.compartment.oc1..aaaaaaaawob4faujxaqxqzrb555b44wxxrfkcpapjxwp4s4hwjthu46idr5a -DLABELING_ALGORITHM=FIRST_LETTER_MATCH -DLABELS=cat,dog -cp libs/bulklabelutility-v2.jar com.oracle.datalabelingservicesamples.scripts.SingleLabelDatasetBulkLabelingScript
+java -DCONFIG_FILE_PATH='~/.oci/config' -DCONFIG_PROFILE=DEFAULT -DDLS_DP_URL=https://dlsprod-dp.<REGION>.oci.oraclecloud.com -DTHREAD_COUNT=20 -DDATASET_ID=ocid1.datalabelingdatasetint.oc1.phx.amaaaaaaniob46ia7ybplfjdfmohqxxmwpg4p6nftl4ypnuirvsljkzhlq3q -DLABELING_ALGORITHM=FIRST_LETTER_MATCH -DLABELS=cat,dog -cp libs/bulklabelutility-v3.jar com.oracle.datalabelingservicesamples.scripts.SingleLabelDatasetBulkLabelingScript
```
Comment (s-sridharan) [inline data_labeling_examples/bulk_labeling_java/README.md:174]: The dp endpoint points to dlsprod but the dataset id says datalabelingdatasetint in the example, might be confusing for users

File: data_labeling_examples/bulk_labeling_java/README.md [hunk 2]
```diff
@@ -125,32 +158,32 @@ java -version
...
 ```
 7. Run the below command to bulk label by "FIRST_REGEX_MATCH" labeling algorithm.
 
 ```
-java -DCONFIG_FILE_PATH='~/.oci/config' -DCONFIG_PROFILE=DEFAULT -DDLS_DP_URL=https://dlsprod-dp.us-ashburn-1.oci.oraclecloud.com -DTHREAD_COUNT=20 -DDATASET_ID=ocid1.compartment.oc1..aaaaaaaawob4faujxaqxqzrb555b44wxxrfkcpapjxwp4s4hwjthu46idr5a -DLABELING_ALGORITHM=FIRST_REGEX_MATCH -DFIRST_MATCH_REGEX_PATTERN=^abc* -DLABELS=cat,dog -cp libs/bulklabelutility-v2.jar com.oracle.datalabelingservicesamples.scripts.SingleLabelDatasetBulkLabelingScript
+java -DCONFIG_FILE_PATH='~/.oci/config' -DCONFIG_PROFILE=DEFAULT -DDLS_DP_URL=https://dlsprod-dp.<REGION>.oci.oraclecloud.com -DTHREAD_COUNT=20 -DDATASET_ID=ocid1.datalabelingdatasetint.oc1.phx.amaaaaaaniob46ia7ybplfjdfmohqxxmwpg4p6nftl4ypnuirvsljkzhlq3q -DLABELING_ALGORITHM=FIRST_REGEX_MATCH -DFIRST_MATCH_REGEX_PATTERN=^abc* -DLABELS=cat,dog -cp libs/bulklabelutility-v3.jar com.oracle.datalabelingservicesamples.scripts.SingleLabelDatasetBulkLabelingScript
```
Comment (s-sridharan) [inline data_labeling_examples/bulk_labeling_java/README.md:179]: The dp endpoint points to dlsprod but the dataset id says datalabelingdatasetint in the example, might be confusing for users

File: data_labeling_examples/bulk_labeling_java/README.md [hunk 3]
```diff
@@ -125,32 +158,32 @@ java -version
...
 ```
 8. Run the below command to bulk label by "CUSTOM_LABELS_MATCH" labeling algorithm.
 
 ```
-java -DCONFIG_FILE_PATH='~/.oci/config' -DCONFIG_PROFILE=DEFAULT -DDLS_DP_URL=https://dlsprod-dp.us-ashburn-1.oci.oraclecloud.com -DTHREAD_COUNT=20 -DDATASET_ID=ocid1.compartment.oc1..aaaaaaaawob4faujxaqxqzrb555b44wxxrfkcpapjxwp4s4hwjthu46idr5a -DLABELING_ALGORITHM=CUSTOM_LABELS_MATCH -DCUSTOM_LABELS='{"dog/": ["dog"], "cat/": ["cat"] }' -cp libs/bulklabelutility-v2.jar com.oracle.datalabelingservicesamples.scripts.CustomBulkLabelingScript
+java -DCONFIG_FILE_PATH='~/.oci/config' -DCONFIG_PROFILE=DEFAULT -DDLS_DP_URL=https://dlsprod-dp.<REGION>.oci.oraclecloud.com -DTHREAD_COUNT=20 -DDATASET_ID=ocid1.datalabelingdatasetint.oc1.phx.amaaaaaaniob46ia7ybplfjdfmohqxxmwpg4p6nftl4ypnuirvsljkzhlq3q -DLABELING_ALGORITHM=CUSTOM_LABELS_MATCH -DCUSTOM_LABELS='{"dog/": ["dog"], "cat/": ["cat"] }' -cp libs/bulklabelutility-v3.jar com.oracle.datalabelingservicesamples.scripts.CustomBulkLabelingScript
```
Comment (s-sridharan) [inline data_labeling_examples/bulk_labeling_java/README.md:184]: The dp endpoint points to dlsprod but the dataset id says datalabelingdatasetint in the example, might be confusing for users

File: data_labeling_examples/bulk_labeling_java/src/main/java/com/oracle/datalabelingservicesamples/scripts/RemoveLabelScript.java [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        List<LabelName> datasetLabelSet = datasetResponse.getDataset().getLabelSet().getItems();
+
+        log.info("Dataset Labels : ");
+        for (LabelName labelName : datasetLabelSet) {
+            actualDatasetLabels.add(labelName.getName());
+            log.info(labelName.getName());
+        }
+        dataset = datasetResponse.getDataset();
+    }
+    private static void processAnnotationSummaryAndRemoveLabel(AnnotationSummary annotationSummary){
```
Comment (s-sridharan) [inline data_labeling_examples/bulk_labeling_java/src/main/java/com/oracle/datalabelingservicesamples/scripts/RemoveLabelScript.java:125]: Requires assertions to prevent NPE during get calls and exception handling

File: data_labeling_examples/bulk_labeling_java/README.md [hunk 5]
```diff
@@ -88,7 +88,40 @@ Conditions -
...
+    dog1.png will be labeled with dog label
+    dog2.png will be labeled with dog and pup labels
+    
+2. If REMOVE_LABEL_PREFIX = 'd' then it will remove label 'dog' from all labeled records. Dataset will be as folows :
+    cat1.jpeg will be labeled with cat label
+    cat2.jpeg will be labeled with kitten labels
+    dog1.png -> unlabeled
+    dog2.png will be labeled with dog and pup labels
+
+3. If REMOVE_LABEL_PREFIX = '*' then it will remove all labels from all labeled records. Dataset will be as folows :
```
Comment (s-sridharan) [inline data_labeling_examples/bulk_labeling_java/README.md:119]: This is a clean method to allow users to delete all annotations!

File: data_labeling_examples/bulk_labeling_java/src/main/java/com/oracle/datalabelingservicesamples/labelingstrategies/FirstLetterMatch.java [hunk 6]
```diff
@@ -11,7 +11,7 @@ public class FirstLetterMatch implements RuleBasedLabelingStrategy {
 	@Override
 	public List<String> getLabel(RecordSummary record) {
 		for (String label : Config.INSTANCE.getLabels()) {
-			if (record.getName().startsWith(label)) {
+			if (record.getName().startsWith(String.valueOf(label.charAt(0)))) {
```
Comment (s-sridharan) [inline data_labeling_examples/bulk_labeling_java/src/main/java/com/oracle/datalabelingservicesamples/labelingstrategies/FirstLetterMatch.java:14]: Why was this change required in the existing functionality?

Comment (mahemaku) [reply data_labeling_examples/bulk_labeling_java/src/main/java/com/oracle/datalabelingservicesamples/labelingstrategies/FirstLetterMatch.java:14]: This change is related to this bug : https://jira.oci.oraclecorp.com/browse/DLS-4883

File: data_labeling_examples/bulk_labeling_java/README.md (hunk 5 above)
Comment (mahemaku) [reply data_labeling_examples/bulk_labeling_java/README.md:119]: yes, it will delete all annotations

File: data_labeling_examples/bulk_labeling_java/README.md (hunk 1 above)
Comment (mahemaku) [reply data_labeling_examples/bulk_labeling_java/README.md:174]: This piece of code is just an example.

File: data_labeling_examples/bulk_labeling_java/src/main/java/com/oracle/datalabelingservicesamples/scripts/RemoveLabelScript.java (hunk 4 above)
Comment (mahemaku) [reply data_labeling_examples/bulk_labeling_java/src/main/java/com/oracle/datalabelingservicesamples/scripts/RemoveLabelScript.java:125]: AnnotationSummary can't be null. 
And after fetching the annotation details using getAnnotation API, we have added the assertion for API response.

**Review by s-sridharan — APPROVED** [review]

**Review by pravemp — APPROVED** [review]
